src/user/user.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Controller, Get, Param } from '@nestjs/common';
+import { UserService } from './user.service';
+import { UserDetails } from './user.datails.interface';
+
+@Controller('user')
+export class UserController {
+  constructor(private userService: UserService) {}
+
+  // if we add the guards then we can't call this method without the token. You can try it :)
+  //@UseGuards(JwtGuard)
+  @Get(':id')
+  getUser(@Param('id') id: string): Promise<UserDetails | null> {
```

Comment:
I think user shouldn't have a controller, we only 'play' with it from the frontend when we want to login or register (in auth controller) so I think it doesn't need a separate controller. 

Instead of getUser maybe we will need getProfile - I made a Profile entity, it only has a username (and id). user is a class where we store passwords and usernames and profiles only have usernames and they have a list of posts, list of saved posts etc. = Profiles have relations with posts and comments - users are used to login and register.

we just need a clever way to connect users and profiles :D maybe they should have the same id